python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -1304,3 +1305,50 @@ def _df_query_method(self, *args, local_dict=None, global_dict=None, **kwargs):
         _Unusable,
         typ,
     )
+
+# timestamps and timedeltas are not proxied and "real" pandas types
+# are currently not picklable when the module accelerator is enabled.
+def _unpickle_timestamp(pickled_args):
+    from cudf.pandas.module_accelerator import disable_module_accelerator
+
+    with disable_module_accelerator():
+        unpickler, args = pickle.loads(pickled_args)
+    ts = unpickler(*args)
+    return ts
+
+
+def _reduce_timestamp(ts):
+    from cudf.pandas.module_accelerator import disable_module_accelerator
+
+    with disable_module_accelerator():
+        # args can contain objects that are unpicklable
+        # when the module accelerator is disabled
+        # (freq is of a proxy type):
+        pickled_args = pickle.dumps(ts.__reduce__())
+
+    return _unpickle_timestamp, (pickled_args,)
+
+
+def _unpickle_timedelta(pickled_args):
```

Comment:
How is `_unpickle_timedelta` different from `_unpickle_timestamp`? Similarly for `_reduce_*`. They look identical to me. Do we need this duplication or can we simplify things?

Reply:
Agreed, this looks like it can be deduplicated.